### PR TITLE
[Federation] Add a note about DNS API scopes to the Federation tutorial

### PR DESCRIPTION
**Pull request by perotinus**
Not sure this is the right place for this, but I wanted to get the text out somewhere before I forgot about it.

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/2943)

### CODE REVIEW

**Comment by perotinus** [comment]
cc @madhusudancs

**Comment by madhusudancs** [comment]
Thanks for sending this update!

/assign @madhusudancs

---

Review status: 0 of 1 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[docs/tutorials/federation/set-up-cluster-federation-kubefed.md, line 110 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes.github.io/2943#-KfmQx1erK8ke0MCydEP:-KfmQx1erK8ke0MCydEQ:bnxq0m8) ([raw file](https://github.com/kubernetes/kubernetes.github.io/blob/a1630248f795c53f550a6e32afdf887046487c0f/docs/tutorials/federation/set-up-cluster-federation-kubefed.md#L110)):*
> ```Markdown
> the `gcloud` command with the appropriate value in the `--scopes` field.
> You cannot modify a GKE cluster directly to add this scope, but you can
> create a new node pool for your cluster and delete the old one:
> ```

It seems a little odd to have this in the note and the command in a regular command block. I think we should just make this a regular paragraph.

---

*[docs/tutorials/federation/set-up-cluster-federation-kubefed.md, line 112 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes.github.io/2943#-KfmQTB6IkubighdvpCv:-KfmQTB6IkubighdvpCw:bn180i5) ([raw file](https://github.com/kubernetes/kubernetes.github.io/blob/a1630248f795c53f550a6e32afdf887046487c0f/docs/tutorials/federation/set-up-cluster-federation-kubefed.md#L112)):*
> ```Markdown
> create a new node pool for your cluster and delete the old one:
> 
> ```bash
> ```

Some docs use `bash` and some use `shell`. `shell` is more common than `bash` and it is at least consistent with other occurrences in this doc. Could you please change that?

---

*[docs/tutorials/federation/set-up-cluster-federation-kubefed.md, line 115 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes.github.io/2943#-KfmUyx3qglIWUj-PXaY:-KfmUyx3qglIWUj-PXaZ:b-aznhqt) ([raw file](https://github.com/kubernetes/kubernetes.github.io/blob/a1630248f795c53f550a6e32afdf887046487c0f/docs/tutorials/federation/set-up-cluster-federation-kubefed.md#L115)):*
> ```Markdown
> gcloud container node-pools create new-np \
>     --cluster gke-cluster \
>     --scopes "...,other_scope,https://www.googleapis.com/auth/ndev.clouddns.readwrite"
> ```

I would really prefer the examples to be copy/paste-able. One way to to do that here is:

```shell
gcloud container node-pools create new-np \
    --cluster gke-cluster \
    --scopes "$(gcloud container node-pools describe --cluster=gl --zone=us-west1-a default-pool --format='value[delimiter=","](config.oauthScopes)'),https://www.googleapis.com/auth/ndev.clouddns.readwrite"
```

OR


```shell
scopes="$(gcloud container node-pools describe --cluster=gl --zone=us-west1-a default-pool --format='value[delimiter=","](config.oauthScopes)')"
gcloud container node-pools create new-np \
    --cluster gke-cluster \
    --scopes "${scopes},https://www.googleapis.com/auth/ndev.clouddns.readwrite"
```

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes.github.io/2943#-:-KfmVYicca-GVZgIWK7z:b-kvg4md)*

**Comment by madhusudancs** [comment]
Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes.github.io/2943)*

**Comment by madhusudancs** [comment]
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[docs/tutorials/federation/set-up-cluster-federation-kubefed.md, line 115 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes.github.io/2943#-KfmUyx3qglIWUj-PXaY:-KfmVldDTfLuQprGdPx8:bue9tgn) ([raw file](https://github.com/kubernetes/kubernetes.github.io/blob/a1630248f795c53f550a6e32afdf887046487c0f/docs/tutorials/federation/set-up-cluster-federation-kubefed.md#L115)):*
<details><summary><i>Previously, madhusudancs (Madhusudan.C.S) wrote…</i></summary><blockquote>

I would really prefer the examples to be copy/paste-able. One way to to do that here is:

```shell
gcloud container node-pools create new-np \
    --cluster gke-cluster \
    --scopes "$(gcloud container node-pools describe --cluster=gl --zone=us-west1-a default-pool --format='value[delimiter=","](config.oauthScopes)'),https://www.googleapis.com/auth/ndev.clouddns.readwrite"
```

OR


```shell
scopes="$(gcloud container node-pools describe --cluster=gl --zone=us-west1-a default-pool --format='value[delimiter=","](config.oauthScopes)')"
gcloud container node-pools create new-np \
    --cluster gke-cluster \
    --scopes "${scopes},https://www.googleapis.com/auth/ndev.clouddns.readwrite"
```
</blockquote></details>

Oh I forgot. In a separate block, also give the command to delete the old node pool?

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes.github.io/2943)*

File: docs/tutorials/federation/set-up-cluster-federation-kubefed.md [hunk 1]
```diff
@@ -97,6 +97,24 @@ kubefed init fellowship --host-cluster-context=rivendell  --dns-zone-name="examp
 The domain suffix specified in `--dns-zone-name` must be an existing
 domain that you control, and that is programmable by your DNS provider.
 
+The machines in your host cluster must have the appropriate permissions
+to program the DNS service that you are using. For example, if your
+cluster is running on Google Compute Engine, you must enable the
+"Google Cloud DNS API" for your project.
```
Comment (chenopis) [inline docs/tutorials/federation/set-up-cluster-federation-kubefed.md]: Style: Google Cloud DNS API doesn't need to be in quotes.

**Comment by perotinus** [comment]
Review status: 0 of 1 files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[docs/tutorials/federation/set-up-cluster-federation-kubefed.md, line 110 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes.github.io/2943#-KfmQx1erK8ke0MCydEP:-Kfr5DJiQui_n8KCj__F:b-896fix) ([raw file](https://github.com/kubernetes/kubernetes.github.io/blob/a1630248f795c53f550a6e32afdf887046487c0f/docs/tutorials/federation/set-up-cluster-federation-kubefed.md#L110)):*
<details><summary><i>Previously, madhusudancs (Madhusudan.C.S) wrote…</i></summary><blockquote>

It seems a little odd to have this in the note and the command in a regular command block. I think we should just make this a regular paragraph.
</blockquote></details>

Done.

---

*[docs/tutorials/federation/set-up-cluster-federation-kubefed.md, line 112 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes.github.io/2943#-KfmQTB6IkubighdvpCv:-Kfr5Di5sdlg7ua4-f7p:b-896fix) ([raw file](https://github.com/kubernetes/kubernetes.github.io/blob/a1630248f795c53f550a6e32afdf887046487c0f/docs/tutorials/federation/set-up-cluster-federation-kubefed.md#L112)):*
<details><summary><i>Previously, madhusudancs (Madhusudan.C.S) wrote…</i></summary><blockquote>

Some docs use `bash` and some use `shell`. `shell` is more common than `bash` and it is at least consistent with other occurrences in this doc. Could you please change that?
</blockquote></details>

Done.

---

*[docs/tutorials/federation/set-up-cluster-federation-kubefed.md, line 115 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes.github.io/2943#-KfmUyx3qglIWUj-PXaY:-Kfr5OmSYbucce0oVefL:bx4nh62) ([raw file](https://github.com/kubernetes/kubernetes.github.io/blob/a1630248f795c53f550a6e32afdf887046487c0f/docs/tutorials/federation/set-up-cluster-federation-kubefed.md#L115)):*
<details><summary><i>Previously, madhusudancs (Madhusudan.C.S) wrote…</i></summary><blockquote>

Oh I forgot. In a separate block, also give the command to delete the old node pool?
</blockquote></details>

Done and done.

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes.github.io/2943)*

**Comment by perotinus** [comment]
Thanks for the comments! PTAL.

File: docs/tutorials/federation/set-up-cluster-federation-kubefed.md (hunk 1 above)
Comment (perotinus) [inline docs/tutorials/federation/set-up-cluster-federation-kubefed.md]: Done.

**Comment by madhusudancs** [comment]
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes.github.io/2943)*

**Comment by perotinus** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[docs/tutorials/federation/set-up-cluster-federation-kubefed.md, line 103 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes.github.io/2943#-KfmELV38eMLxWJ7gNjH-r1-103:-Kfru3mu8lbNNz_CewyD:b-896fix) ([raw file](https://github.com/kubernetes/kubernetes.github.io/blob/a1630248f795c53f550a6e32afdf887046487c0f/docs/tutorials/federation/set-up-cluster-federation-kubefed.md#L103)):*
<details><summary><i>Previously, perotinus (Jonathan MacMillan) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes.github.io/2943)*

**Pull request merged**